src/journal/ObjectRecorder.cc

```diff
@@ -295,8 +300,18 @@ void ObjectRecorder::handle_append_flushed(uint64_t tid, int r) {
 
   if (m_in_flight_appends.empty() && !m_aio_scheduled && m_object_closed) {
     // all remaining unsent appends should be redirected to new object
+    m_append_buffers.splice(m_append_buffers.begin(), m_pending_buffers);
     notify_handler_unlock();
   } else {
+    if (!m_aio_scheduled && !m_pending_buffers.empty() &&
```

Comment:
Since this check is already in ```send_appends_aio```, can we remove it here and take the small hit that something gets scheduled that potentially turn into a no-op just to simplify it a little bit?